### PR TITLE
Refactor Logging System for Enhanced Debugging and Clarity

### LINKED ISSUE #114935: print statements used in torch/utils/cpp_extension.py
### 🐛 Describe the bug

There are a lot of `print()` statements in `torch/utils/cpp_extension.py` ([here](https://github.com/pytorch/pytorch/blob/4875e4d63fa3e64c4fcd88c079fc7699b3e89d74/torch/utils/cpp_extension.py#L2034) for example). I don't think this is good practice. Maybe we should consider the standard `logging` package instead?

### Versions

Irrelevent

### PR DESCRIPTION
**Pull request by RazaProdigy**
Fixes #114935

### Key Changes:

1. Introduction of Logging Module: Implemented Python's built-in logging module, replacing existing print statements used for debugging and error reporting.
2. Logging Level and Format Configuration: Configured the logging system with an appropriate level (INFO) and a standard format that includes timestamps and log levels for easier tracking and readability of log messages.
3. Error and Information Handling: Transformed all critical error messages to use logging.error and standard information outputs to logging.info, ensuring a clear distinction between different types of log messages.

### Impact of Changes:

- Improved Debugging: Easier identification and troubleshooting of issues with detailed, timestamped log messages.
- Code Cleanliness and Consistency: A more standardized approach to handling logs and errors throughout the codebase.
- Scalability for Future Development: Establishes a robust foundation for expanding the logging system with additional levels or formats as the application evolves.

I look forward to the team's review and feedback on this enhancement. Thank you!

### CODE REVIEW

File: torch/utils/cpp_extension.py [hunk 1]
```diff
@@ -26,6 +27,9 @@
 
 from setuptools.command.build_ext import build_ext
 
+# Configure logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
```
Comment (ezyang) [inline torch/utils/cpp_extension.py:31]: this is not allowed

**Comment by ezyang** [comment]
It is reasonable to make these logs, but it is not ok to change log setting on module import. But unfortunately, there isn't really a good place to change the log setting since we suppress INFO by default. Maybe we should just make them all warnings

**Comment by RazaProdigy** [comment]
@ezyang Understood. Committed the changes. Could you please have a look again? Thank you!

File: torch/utils/cpp_extension.py [hunk 2]
```diff
@@ -26,6 +27,12 @@
 
 from setuptools.command.build_ext import build_ext
 
+# Create a logger for this module
+log = logging.getLogger(__name__)
+
+# Set the default log level to WARNING to prevent info messages from being displayed
+log.setLevel(logging.WARNING)
```
Comment (ezyang) [inline torch/utils/cpp_extension.py:34]: no you don't need thi

**Review by ezyang — CHANGES_REQUESTED** [review]

**Pull request closed without merging**